test/integration/globals/app-nav.js

```diff
@@ -210,6 +210,98 @@ context('App Nav', function() {
       .should('not.have.class', 'is-selected');
   });
 
+  specify('switch workspaces', function() {
+    cy
+      .routeActions()
+      .visit()
+      .wait('@routeActions')
+      .its('request.headers')
+      .should('have.property', 'workspace', '11111')
+      .then(() => {
+        const storage = JSON.parse(localStorage.getItem('currentWorkspace'));
+
+        expect(storage).to.equal('11111');
+      });
+
+    cy
+      .url()
+      .should('contain', '/one/worklist/owned-by');
+
+    cy
+      .get('.app-nav__header')
+      .as('mainNav')
+      .click();
+
+    cy
+      .get('.picklist')
+      .find('.picklist__group')
+      .find('.picklist__item')
+      .first()
+      .should('have.class', 'is-selected')
+      .next()
+      .should('not.have.class', 'is-selected')
+      .click();
+
+    cy
+      .wait('@routeActions')
+      .its('request.headers')
+      .should('have.property', 'workspace', '22222')
+      .then(() => {
+        const storage = JSON.parse(localStorage.getItem('currentWorkspace'));
+
+        expect(storage).to.equal('22222');
+      });
+
+    cy
+      .url()
+      .should('contain', '/two/worklist/owned-by');
+
+    cy
+      .get('.app-nav__header')
+      .as('mainNav')
+      .click();
+
+    cy
+      .get('.picklist')
+      .find('.picklist__group')
+      .find('.picklist__item')
+      .first()
+      .should('not.have.class', 'is-selected')
+      .next()
+      .should('have.class', 'is-selected');
+
+    cy
+      .go('back');
+
+    cy
+      .wait('@routeActions')
+      .its('request.headers')
+      .should('have.property', 'workspace', '11111')
+      .then(() => {
+        const storage = JSON.parse(localStorage.getItem('currentWorkspace'));
+
+        expect(storage).to.equal('11111');
+      });
+
+    cy
+      .url()
+      .should('contain', '/one/worklist/owned-by');
+
+    cy
+      .get('.app-nav__header')
+      .as('mainNav')
+      .click();
+
+    cy
+      .get('.picklist')
+      .find('.picklist__group')
+      .find('.picklist__item')
+      .first()
+      .should('have.class', 'is-selected')
+      .next()
+      .should('not.have.class', 'is-selected');
+  });
+
```

Comment:
Checks that the page URL, workspace request header, workspace local storage value, and CSS active states of the droplist items are set properly when switching between workspaces via the app nav menu.